packages/next-server/server/next-server.ts

```diff
@@ -93,8 +101,11 @@ export default class Server {
       publicRuntimeConfig,
     })
 
-    const routes = this.generateRoutes()
-    this.router = new Router(routes)
+    this.router = new Router()
+    this.routesPromise = this.generateRoutes().then((routes) => {
```

Comment:
This actually makes Next.js serve 404 on bootup 🤔 It should be done in `prepare` imo.